draft-ietf-ppm-dap.md

```diff
@@ -2878,33 +2948,140 @@ struct {
 computed above and `encrypted_aggregate_share.ciphertext` is the ciphertext
 `encrypted_agg_share` computed above.
 
-The Helper's handling of this request MUST be idempotent. That is, if multiple
-identical, valid `AggregateShareReq`s are received, they should all yield the
-same response.
+The Helper MAY defer handling the aggregate share request. In this case, it
+indicates that the response is not yet ready by immediately sending an empty
+response body. The Helper SHOULD include a Retry-After header field to suggest
+a polling interval to the Leader. The Leader then polls the state of the job by
+sending GET requests to the aggregate share. The Helper responds the same way
+until the job is ready, from which point it responds with the encoded
+`AggregateShare`. The Leader SHOULD use each response's Retry-After header field
+to decide when to try again.
+
+The Helper MAY also block responding to the Leader's PUT until the aggregate
+share is ready. In this case, it responds with the `AggregateShare`.
 
-After receiving the Helper's response, the Leader uses the HpkeCiphertext to
-finalize a collection job (see {{collect-finalization}}).
+After receiving the Helper's response, the Leader includes the HpkeCiphertext in
+its response to the Collector (see {{collect-finalization}}).
```

Comment:
I found it awkward to imply that the Leader uses the Helper's agg share for anything, as well as the implication that the leader finalizes the collection job. Seems like that happens when the collector unshards the agg shares.